react/components/EXPERIMENTAL_Table/README.md

```diff
@@ -642,6 +642,1091 @@ function VisibilityExample() {
+        ...state,
+        inputValue: value,
+      }
+    }
+    case 'clear': {
+      return {
+        inputValue: '',
+        displayItems: items,
+      }
+    }
+    case 'submit': {
+      const inputClear = state.input === ''
+      const filterFn = item =>
+        item.name.toLowerCase().includes(state.inputValue.toLowerCase())
+      return {
+        ...state,
+        displayItems: inputClear ? items : items.filter(filterFn),
+      }
+    }
+    default: {
+      return state
+    }
+  }
+}
+
+function ToolbarExample() {
+  const [state, dispatch] = React.useReducer(reducer, {
+    inputValue: '',
+    displayItems: items,
+  })
+
+  const visibility = useTableVisibility({
+    columns,
+    items,
+    hiddenColumns: ['id'],
+  })
+
+  const measures = useTableMeasures({
+    size: state.displayItems.length,
+  })
+
+  const inputSearch = {
+    value: state.inputValue,
+    placeholder: 'Search stuff...',
+    onChange: e => dispatch({ type: 'change', value: e.currentTarget.value }),
+    onClear: () => {
+      dispatch({ type: 'clear' })
+    },
+    onSubmit: e => {
+      e.preventDefault()
+      dispatch({ type: 'submit' })
+    },
+  }
+
+  const buttonColumns = {
+    label: 'Toggle visible fields',
+    showAllLabel: 'Show All',
+    hideAllLabel: 'Hide All',
+    visibility,
+  }
+
+  const density = {
+    label: 'Line density',
+    compactLabel: 'Compact',
+    regularLabel: 'Regular',
+    comfortableLabel: 'Comfortable',
+  }
+
+  const download = {
+    label: 'Export',
+    onClick: () => alert('Clicked EXPORT'),
+  }
+
+  const upload = {
+    label: 'Import',
+    onClick: () => alert('Clicked IMPORT'),
+  }
+
+  const extraActions = {
+    label: 'More options',
+    actions: [
+      {
+        label: 'An action',
+        onClick: () => alert('An action'),
+      },
+      {
+        label: 'Another action',
+        onClick: () => alert('Another action'),
+      },
+      {
+        label: 'A third action',
+        onClick: () => alert('A third action'),
+      },
+    ],
+  }
+
+  const newLine = {
+    label: 'New',
+    onClick: () => alert('handle new line callback'),
+    actions: [
+      'General',
+      'Desktop & Screen Saver',
+      'Dock',
+      'Language & Region',
+    ].map(label => ({
+      label,
+      onClick: () => alert(`Clicked ${label}`),
+    })),
+  }
+
+  const emptyState = {
+    label: 'The table is empty',
+  }
+
+  const empty = React.useMemo(
+    () =>
+      state.displayItems.length === 0 ||
+      Object.keys(visibility.visibleColumns).length === 0,
+    [visibility.visibleColumns, state.displayItems]
+  )
+
+  return (
+    <Table
+      empty={empty}
+      measures={measures}
+      items={state.displayItems}
+      columns={visibility.visibleColumns}
+      emptyState={emptyState}>
+      <Table.Toolbar>
+        <Table.Toolbar.InputSearch {...inputSearch} />
+        <Table.Toolbar.ButtonGroup>
+          <Table.Toolbar.ButtonGroup.Columns {...buttonColumns} />
```

Comment:
If we click on the row it works, but if we click on the toggle it does not change the value.

Reply:
This is a bug of the component. Since this is a documentation PR, I'll do another fixing it.